### ISSUE TITLE
Column Numbering change - In Center of botton panel.

### ISSUE THREAD

**Issue by IanMcGilp** [notepad-plus-plus/notepad-plus-plus#5320]
### Description of the Issue
When you start counting the letters/columns/spaces the one after a carriage return should be zero and not 1 as you have not typed a character or moved a space yet. 

### Steps to Reproduce the Issue

1.  Open notepad++
2. Look at the bottom panel 
3. Col: 1

### Expected Behavior

Col: 0 
### Actual Behavior

Col: 1
### Debug Information

This should be an easy fix, low hanging fruit.

**Comment by sasumner** [comment]
It _would_ be an easy fix but it isn't broken.  
Works as intended.
Please close this issue.

**Comment by Faq** [comment]
@sasumner  explain more what you mean it is not broken. So why 

> Col: 1

 is valid?

**Comment by sasumner** [comment]
There really is no explanation; it is just "how it works".  Maybe the best in terms of an explanation is that the user never sees zero as a line number or a column number.  [But definitely under the hood Notepad++ and Scintilla manipulate row/column data on a zero-indexed basis.]

Perhaps the difficulty in interpretation happens because an "insert" mode caret appears between characters.

**Comment by IanMcGilp** [comment]
I do understand about the ^ and $ etc. But then when you go to view all characters, why is this not there? Even though you cannot see it, treat it like a carriage return or something similar?

I am just saying that for example, I am working on a FedEx project and counting characters is important.
When your average bear starts counting and finds that column one is shown as column two, that does not engender confidence in the product. 

What your doing is kind of like saying that a baby has taken it's first step when it has stood up and not even moved a foot, however common sense, which does not appear to be as common as one might think, is indeed absent as it did not even take a step!

So if you are intent on being so correct, show the ^ at the start of a line to at least acknowledge it in the show all characters?  You are already handling regular expressions anyhow!

**Comment by sasumner** [comment]
Do people feel better about this if they either (a) press the Ins key such that the caret turns into an underbar, or (b) change the caret to a "Block" in the settings?

>counting characters is important

Why not "select" characters to count them? At that point the status bar will show how many are "Sel" ected.

**Comment by IanMcGilp** [comment]
Because it is the characters position (Column number) that you are looking for like where the 021 is in this example: NNNNNNNNO000000000000000000000000210000000000000000000000000000GR.

**Comment by IanMcGilp** [comment]
Selecting does give the correct number, But this does not let you off the hook. Looks like you might need some help from a code guru? Pass it on to a person more qualified. Selecting the text is but a work-around.

**Comment by sasumner** [comment]
[Okay, well that is a bit different than the "counting of characters" I thought you meant; so I retract my suggestion to "select" characters for your use case.]

Did you even try my suggestions labelled "(a)" and "(b)" above?

So as a setup to what comes below I copy your NNNN.....GR string above to an empty Notepad++ tab window....

If press the "Ins" once and release it to put myself into OVR mode (see right hand side of Notepad++'s status bar), as opposed to it showing INS like it did before I pressed the Ins key...and then I navigate the (underbar) caret to be under the `0` in your `021` example, I see 33 as the column it is in.  Seems correct to me and even you have to like that.

If I then press and release the "Ins" key again (now I see INS on status bar), the caret (with a caret width setting = 3) now blinks BEFORE (aka to the left of) the `0` in `021` and the column indicator has not changed; still showing 33.  This also seems correct to me, as any new characters will get inserted BETWEEN existing characters, right where the caret is.  I'm perfectly happy with knowing that column 33 is to the RIGHT of my blinking caret.

Now if I go into the Settings / Preferences and set my Caret Width to "Block", I am again still in column 33 but now my caret is blinking right squarely over the `0` in `021` -- again, even you can't argue with that.

So, in short, there is no bug here, and really no complaint.  You just have to understand how text editors work.  Every single text editor I've used over 35 years of code development (I "might need some help from a code guru"--really?).  Even--and I hate to say it because I don't like citing it as an example of proper behavior--Microsoft Visual Studio's (ugh!) text editor works this same way (never shows a column zero).

Thus I repeat my original closing statement: Please close this issue.

**Comment by DavidHansen-0x454d43** [comment]
I'm with sasumner on this one. Its working as expected and like every other text editor I've seen. I think the confusion comes from the default behavior of insert mode, where the vertical bar is _in between_ columns. It becomes much clearer if you drop into overwrite mode (by toggling the Insert key). There is no ambiguity of which column your cursor is in when you're in overwrite mode.

**Comment by MetaChuh** [comment]
the current numbering is by design and will remain.

thank you for your help and best regards.

-----
```
general notes:

please close your issues as soon as you are satisfied, to help
keeping the notepad++ issue tracker clean.

when creating a new issue, please provide all information as shown at the issue template.
your notepad++ debug info is part of the minimum requirements.
(? > Debug Info... > Copy debug info into clipboard).

please also conduct a search at the issue tracker beforehand, to check if a similar issue
already exists.

for general questions, or if it is not sure, whether your issue is directly related to
the notepad++ source code, please visit us at the notepad++ community forum, and
search if related topics exists. you are welcome to post either at similar
topics, or to create a new topic at Help Wanted or General Discussion.
(no extra account is needed, just use your github account to sign in)
```
[<p align="center"><kbd> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; Click here to visit the Notepad++ Community Forum &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;&emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; </kbd></p>](https://notepad-plus-plus.org/community/)
